Dockerfile.sync_envs_with_prod

```diff
@@ -0,0 +1,17 @@
+ARG BASE_TAG=branch-main
```

Comment:
re: file name, we're using both "mirror" and "sync" terms and we should probably settle on one for consistency amongst all related code.  For the load lambda, I was going to call it `mirror-prod-env-data-to-<env>`, e.g.  I don't feel strongly about any particular name, just the consistency. :)

Reply:
(we can circle back on naming at the end, once we have a working system.)